swift/apple/FirezoneNetworkExtension/Adapter.swift

```diff
@@ -139,85 +146,102 @@ class Adapter {
     }
   }
 
-  /// Currently disabled resources
+  /// Internet resource enabled state
   private var internetResourceEnabled: Bool
 
-  /// Cache of internet resource
-  private var internetResource: Resource?
-
-  /// Keep track of resources
+  /// Keep track of resources for UI
   private var resourceListJSON: String?
 
   /// Starting parameters
-  private let apiURL: String
+  private let apiUrl: String
   private let token: Token
-  private let id: String
+  private let deviceId: String
+  private let accountSlug: String
   private let logFilter: String
-  private let connlibLogFolderPath: String
 
-  init(
-    apiURL: String,
-    token: Token,
-    id: String,
-    logFilter: String,
-    accountSlug: String,
-    internetResourceEnabled: Bool,
-    packetTunnelProvider: PacketTunnelProvider
-  ) {
-    self.apiURL = apiURL
-    self.token = token
-    self.id = id
-    self.packetTunnelProvider = packetTunnelProvider
-    self.callbackHandler = CallbackHandler()
-    self.logFilter = logFilter
-    self.accountSlug = accountSlug
-    self.connlibLogFolderPath = SharedAccess.connlibLogFolderURL?.path ?? ""
-    self.networkSettings = nil
-    self.internetResourceEnabled = internetResourceEnabled
-  }
-
-  // Could happen abruptly if the process is killed.
-  deinit {
-    Log.log("Adapter.deinit")
+  func start() throws {
+    Log.log("Adapter.start: Starting session for account: \(accountSlug)")
 
-    // Cancel network monitor
-    networkMonitor?.cancel()
-  }
+    // Get device metadata
+    let deviceName = DeviceMetadata.getDeviceName()
+    let osVersion = DeviceMetadata.getOSVersion()
+    let deviceInfo = try JSONEncoder().encode(DeviceMetadata.deviceInfo())
+    let deviceInfoStr = String(data: deviceInfo, encoding: .utf8) ?? "{}"
+    let logDir = SharedAccess.connlibLogFolderURL?.path ?? "/tmp/firezone"
 
-  /// Start the tunnel.
-  func start() throws {
-    Log.log("Adapter.start")
+    // Create the session
+    let session: Session
+    do {
+      session = try Session.newApple(
+        apiUrl: apiUrl,
+        token: token.description,
+        deviceId: deviceId,
+        accountSlug: accountSlug,
+        deviceName: deviceName,
+        osVersion: osVersion,
+        logDir: logDir,
+        logFilter: logFilter,
+        deviceInfo: deviceInfoStr,
+        isInternetResourceActive: internetResourceEnabled
+      )
+    } catch {
+      throw AdapterError.connlibConnectError(String(describing: error))
+    }
 
-    guard session == nil else {
-      throw AdapterError.invalidSession(session)
+    do {
+      try session.setTunFromSearch()
```

Comment:
Can we just call `Tun::new` inside `Session.newApple` on the Rust side?

Eventually (when we build a headless Apple client), we might want to change that but for this PR, it seems simpler to stick to what we are currently doing because it means we don't have to introduce this `setTunFromSearch` function.

Reply:
yes, we could do it on the Rust side. I've kept the function with the retry logic there though, I just call it directly now. 

the main footgun around here was the fact that I needed a tokio runtime for `Tun::new()` - and it was pretty nasty, because the network extension crashing due to that is not presented well in our main app (looks we're just being signed out).

I've added explicit passing of the runtime for both Android and Swift to self-document this requirement.